webapp/templates/login.html

```diff
@@ -13,14 +13,66 @@
       href="{{ .BaseURL }}/assets/styles.<%= webpack.hash %>.css"
     />
     <link rel="icon" href="{{ .BaseURL }}/assets/images/favicon.ico" />
+    <script>
+      function login() {
+        var xhr = new XMLHttpRequest();
+        xhr.open("POST", "/login");
+        xhr.setRequestHeader("Accept", "application/json");
+        xhr.setRequestHeader("Content-Type", "application/json");
+  
+        xhr.onreadystatechange = function () {
+          if (xhr.readyState === 4) {
+            if ( xhr.status == 204 ) {
+              location.href='/';
+            } else {
+              error(xhr.responseText);
+            }
+          }};
+        xhr.send(JSON.stringify({
+          username: document.getElementById('username').value,
+          password: btoa(unescape(encodeURIComponent(document.getElementById('password').value))),
+        }));
+      }
+
+      function error(string) {
+        document.getElementById('error').innerHTML = string;
+      }
+    </script>
   </head>
 
   <body class="login-page">
     <div class="bg-2"></div>
     <section class="login-form">
       <div class="welcome-logo"></div>
       <h1 class="welcome-title">Welcome to Pyroscope</h1>
+      <form>
+        <div class="signin-form">
+          <div class="error" id="error"></div>
+          <div>
+            <div><label>Email or username</label></div>
+            <input type="text" name="user" autocapitalize="none" placeholder="email or username" id="username" />
+          </div>
+          <div>
+            <div><label>Password</label></div>
+            <input type="password" name="password" aria-label="password input field" id="password" autocomplete="current-password"  placeholder="password" />
+          </div>
+          <div>
+            <button type="button" class="sign-in-button" onClick="login()">Log In</button>
```

Comment:
I'd move `login` function to `form onSubmit`, and make the button `type=submit`. that way it works when people press enter and things like that.

Reply:
Sure, I'd change all other forms same way.